packages/blocks/src/root-block/edgeless/edgeless-root-service.ts

```diff
@@ -83,8 +84,8 @@ export class EdgelessRootService extends RootService {
       blockId?: string;
     }>(),
     tagClicked: new Slot<{ tagId: string }>(),
-    editorModeSwitch: new Slot<'edgeless' | 'page'>(),
-
+    editorModeSwitch: new Slot<DocMode>(),
+    setEditorMode: new Slot<{ id: string; mode: DocMode }>(),
```

Comment:
I think `slots.editorModeSwitch` in base `RootService` should also be amended.
BTW IMO we can reuse & extend `editorModeSwitch`'s event so that there is no need to introduce another event slot.

Reply:
I used to think it's a good idea to separate set doc mode into two parts, set current doc and set others, because there are lots of listeners observe current doc mode change.

Reply:
Let me check the releated logic.